### DISCUSSION TITLE
Distributing PHAR with Composer

### DISCUSSION THREAD

**Discussion by zonuexe** [phpactor/phpactor#2308] [Ideas]
Thank you for the release of 2023.06.17. I especially like the idea of distributing PHARs.

This PHAR file can be redistributed in a Composer package so that users can easily install it.

* [psalm/phar](https://github.com/psalm/phar) and [phpDocumentor/shim](https://github.com/phpDocumentor/shim) provide PHAR-based packages.
* [PHPStan](https://packagist.org/packages/phpstan/phpstan) and [Rector](https://packagist.org/packages/rector/rector) only publish PHAR versions of their packages to Packagist.
* Rector author Tomas blogs about [publishing files to Packagist based on a single Git repository](https://tomasvotruba.com/blog/how-to-release-php-81-and-72-package-in-the-same-repository).
* There is also a tool called [phar-io/composer-distributor](https://github.com/phar-io/composer-distributor) that I haven't used.
* I tried https://github.com/zonuexe/phpactor-dist and it seems to work fine.

Are there any plans to provide such a package officially?

Thanks to Dan and all the contributors.

**Comment by dantleech** [comment]
Hey -- so what would be the idea? `composer install --global phpactor/phpactor-dist` ?

I haven't been a fan of compose global installs due to dependency conflicts, but the PHAR avoids this, and the advantage is it's easy to upgrade I guess?

**Comment by zonuexe** [comment]
[`zonuexe/phpactor-dist`](https://packagist.org/packages/zonuexe/phpactor-dist) package can actually install tools without dependencies.

Users install or update Phpactor with the following command:

```
composer global require zonuexe/phpactor-dist
```

If necessary, add `$HOME/.composer/bin` to `PATH` to make it available as a command.

**Comment by dantleech** [comment]
I think this solves the "upgrade" issue at the expense of supporting another installation method, I guess we'd need to setup a new repo and publish the PHAR to it when tagging a new release, which is fairly easy...
